From a69261b9757a0dcf456439a391af470285770ff1 Mon Sep 17 00:00:00 2001
From: George Tankersley <george.tankersley@gmail.com>
Date: Wed, 19 Nov 2014 11:54:18 -0800
Subject: [PATCH] locate crt0-efi-x86_64.o correctly

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 332a29b..b8359e7 100644
--- a/Makefile
+++ b/Makefile
@@ -10,7 +10,7 @@ LIB_PATH	= /usr/lib64
 
 EFI_INCLUDE	:= /usr/include/efi
 EFI_INCLUDES	= -nostdinc -ICryptlib -ICryptlib/Include -I$(EFI_INCLUDE) -I$(EFI_INCLUDE)/$(ARCH) -I$(EFI_INCLUDE)/protocol -Iinclude
-EFI_PATH	:= /usr/lib64/gnuefi
+EFI_PATH	:= $(shell find /usr/lib* -name "crt0-efi-x86_64.o" -exec dirname {} \;)
 
 LIB_GCC		= $(shell $(CC) -print-libgcc-file-name)
 EFI_LIBS	= -lefi -lgnuefi --start-group Cryptlib/libcryptlib.a Cryptlib/OpenSSL/libopenssl.a --end-group $(LIB_GCC) 
-- 
1.9.1

